### PR TITLE
jq: 1.5 -> 1.6 (!!!!)

### PR DESCRIPTION
**Pull request by dtzWill**
###### Motivation for this change

WOOOOOOO

Also, no longer requires specifying '.' when using in a chain/piping it!!

Unsure impact on rebuilds, and looks to be used in a few scripts in-tree
which should be checked to work as expected.

But first, let's see what borg says :).

(oh and I checked commit history and there are indeed commits that (claim)
to fix the issue we had the Darwin strptime patch for)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by dtzWill** [comment]
In-tree uses of the utility appear to be mostly in update scripts
which IMO are lower priority than build breakage or broken packages :).

**Comment by Profpatsch** [comment]
cc @aszlig @fpletz

File: pkgs/development/tools/jq/default.nix [hunk 1]
```diff
@@ -2,48 +2,25 @@
...
+  version="1.6";
 
   src = fetchurl {
     url="https://github.com/stedolan/jq/releases/download/jq-${version}/jq-${version}.tar.gz";
-    sha256="0g29kyz4ykasdcrb0zmbrp2jqs9kv1wz9swx849i2d1ncknbzln4";
+    sha256="1a76f46a652i2g333kfvrl6mp2w7whf6h1yly519izg4y967h9cn";
   };
 
-  outputs = [ "bin" "doc" "man" "dev" "lib" "out" ];
+  outputs = [ "out" "doc" "man" "dev" "lib" ];
```
Comment (Profpatsch) [inline pkgs/development/tools/jq/default.nix]: `bin` should be the first output, the original was as intended.

File: pkgs/development/tools/jq/default.nix [hunk 2]
```diff
@@ -2,48 +2,25 @@
...
-    ++ stdenv.lib.optional (!stdenv.isDarwin) "LDFLAGS=-Wl,-rpath,\\\${libdir}";
+  # jq is linked to libjq:
+  configureFlags = stdenv.lib.optional (!stdenv.isDarwin) "LDFLAGS=-Wl,-rpath,\\\${libdir}";
 
   doInstallCheck = true;
   installCheckTarget = "check";
 
   postInstallCheck = ''
-    $bin/bin/jq --help >/dev/null
+    $out/bin/jq --help >/dev/null
```
Comment (Profpatsch) [inline pkgs/development/tools/jq/default.nix]: As above, this should stay `bin`.

File: pkgs/development/tools/jq/default.nix [hunk 3]
```diff
@@ -2,48 +2,25 @@
...
-      url = https://github.com/stedolan/jq/commit/8eb1367ca44e772963e704a700ef72ae2e12babd.diff;
-      sha256 = "18bjanzvklfzlzzd690y88725l7iwl4f6wnr429na5pfmircbpvh";
-    })
-    (fetchpatch {
-      name = "CVE-2016-4074.patch";
-      url = https://patch-diff.githubusercontent.com/raw/stedolan/jq/pull/1214.diff;
-      sha256 = "1w8bapnyp56di6p9casbfczfn8258rw0z16grydavdjddfm280l9";
-    })
-  ]
-    ++ stdenv.lib.optional stdenv.isDarwin ./darwin-strptime-test.patch;
```
Comment (Profpatsch) [inline pkgs/development/tools/jq/default.nix]: Are you sure these patches can just be removed for 1.6?

**Review by Profpatsch — CHANGES_REQUESTED** [review]
The bin changes should be reverted, the rationale is somewhere in the manual.

**Comment by Profpatsch** [comment]
From the changelog [here](https://github.com/stedolan/jq/releases/tag/jq-1.6) it sounds like the new version should be backwards compatible with 1.5?

**Comment by dtzWill** [comment]
On Fri, 02 Nov 2018 16:06:57 -0700, Profpatsch <notifications@github.com> wrote:
> Profpatsch requested changes on this pull request.
> 
> The bin changes should be reverted, the rationale is somewhere in the manual.
> 
> >    };
>  
> -  outputs = [ "bin" "doc" "man" "dev" "lib" "out" ];
> +  outputs = [ "out" "doc" "man" "dev" "lib" ];
> 
> `bin` should be the first output, the original was as intended.
> 
> >  
>    doInstallCheck = true;
>    installCheckTarget = "check";
>  
>    postInstallCheck = ''
> -    $bin/bin/jq --help >/dev/null
> +    $out/bin/jq --help >/dev/null
> 
> As above, this should stay `bin`.

I thought I remembered there being a reason
but didn't think to check the manual, whoops.

Will refresh without these two changes shortly.

> 
> >  
>    buildInputs = [ oniguruma ];
>  
> -  patches = [
> -    (fetchpatch {
> -      name = "CVE-2015-8863.patch";
> -      url = https://github.com/stedolan/jq/commit/8eb1367ca44e772963e704a700ef72ae2e12babd.diff;
> -      sha256 = "18bjanzvklfzlzzd690y88725l7iwl4f6wnr429na5pfmircbpvh";
> -    })
> -    (fetchpatch {
> -      name = "CVE-2016-4074.patch";
> -      url = https://patch-diff.githubusercontent.com/raw/stedolan/jq/pull/1214.diff;
> -      sha256 = "1w8bapnyp56di6p9casbfczfn8258rw0z16grydavdjddfm280l9";
> -    })
> -  ]
> -    ++ stdenv.lib.optional stdenv.isDarwin ./darwin-strptime-test.patch;
> 
> Are you sure these patches can just be removed for 1.6? 


I didn't thoroughly investigate but they both are from upstream
and seem merged (and not reverted) so I'd say it's safe.

The Darwin patch I mention in the PR text, or I meant to :).

**Comment by dtzWill** [comment]
Honestly it'd be surprising if it wasn't, but first release in X years
made it seem worth at least discussing.  Glad it's compat! :)

**Review by Profpatsch — APPROVED** [review]

**Comment by Profpatsch** [comment]
Looking good!

**Pull request merged**